### ISSUE TITLE
Android - Create Chat - "You appear to be offline"  is displayed floating on room creation

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54157]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.76-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5347500&group_by=cases:section_id&group_order=asc&group_id=229067
**Email or phone of affected tester (no customers):** ibellicotest+91@gmail.com
**Issue reported by:** Applause Internal Team
**Device used:** Motorola MotoG60 - Android 12
**App Component:** Other

## Action Performed:
1. Open the Expensify app.
2. Turn off internet connection.
3. Tap on FAB and select "Start Chat"
4. Tap on "Room"
5. Tap on the arrow on the top left corner.
6. Tap on FAB again.
7. Wait for the keyboard to open.
8. Tap on "Visibility"

## Expected Result:
"You appear to be offline" message should be displayed at the bottom of the screen.

## Actual Result:
"You appear to be offline" message is displayed floating almost in the middle of the screen, when entering "Visibility" section while creating a room.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/5507802e-b409-49e6-ab61-78a700e78efa
![Screenshot_20241214-143227](https://github.com/user-attachments/assets/96f4ccf7-b924-4e93-a867-98c2969d91fd)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- "You appear to be offline" message is displayed floating almost in the middle of the screen, when entering "Visibility" section while creating a room.

### What is the root cause of that problem?

- The issue appears when we are on room page, open the keyboard and then open another screen/modal. And, it is only countered in android.

- We are using keyboard avoiding view in [value picker modal](https://github.com/Expensify/App/blob/9671e6c9d68c53bbf755bb83cecdb35d0c23938b/src/components/ValuePicker/ValueSelectorModal.tsx#L28), which will automatically adjust the screen height when there is the keyboard. And this issue appears because of [KeyboardAvoidingView/index.android.tsx](https://github.com/Expensify/App/blob/main/src/components/KeyboardAvoidingView/index.android.tsx).

- The original bug/problem comes from the fact, that [useKeyboardHandler](https://github.com/Expensify/App/blob/b80c68ea76cdf825d9331bbf298cc41b40716037/src/components/KeyboardAvoidingView/index.android.tsx#L20) in that file is attaching handlers asynchronously (it's attaching it on UI thread) and there could be a situation, when we start to mount a handler while keyboard is visible, then keyboard gets closed and we couldn't register a handler yet. In this case we'll have a bug that described in the issue - we see a keyboard space even if keyboard is already closed. As a result, the offline indicator is displayed floating on room creation.


### What changes do you think we should make in order to solve the problem?
- We don't need to wrap a view in KeyboardAvoidingView if a keyboard will never appear there. So just need to update this:
https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/components/ValuePicker/ValueSelectorModal.tsx#L29
to:


```
            <ScreenWrapper
                shouldEnableKeyboardAvoidingView={false}
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- "You appear to be offline" message is displayed floating almost in the middle of the screen, when entering "Visibility" section while creating a room.

### What is the root cause of that problem?

- The issue appears when we are on room page, open the keyboard and then open another screen/modal. And, it is only countered in android.

- We are using keyboard avoiding view in [value picker modal](https://github.com/Expensify/App/blob/9671e6c9d68c53bbf755bb83cecdb35d0c23938b/src/components/ValuePicker/ValueSelectorModal.tsx#L28), which will automatically adjust the screen height when there is the keyboard. And this issue appears because of [KeyboardAvoidingView/index.android.tsx](https://github.com/Expensify/App/blob/main/src/components/KeyboardAvoidingView/index.android.tsx).

- The original bug/problem comes from the fact, that [useKeyboardHandler](https://github.com/Expensify/App/blob/b80c68ea76cdf825d9331bbf298cc41b40716037/src/components/KeyboardAvoidingView/index.android.tsx#L20) in that file is attaching handlers asynchronously (it's attaching it on UI thread) and there could be a situation, when we start to mount a handler while keyboard is visible, then keyboard gets closed and we couldn't register a handler yet. In this case we'll have a bug that described in the issue - we see a keyboard space even if keyboard is already closed. As a result, the offline indicator is displayed floating on room creation.


### What changes do you think we should make in order to solve the problem?

- We don't need to wrap a view in KeyboardAvoidingView if a keyboard will never appear there
- In [ValueSelectorModal](https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/components/ValuePicker/ValueSelectorModal.tsx#L29), introduce new param, ```shouldEnableKeyboardAvoidingView```, default is ```true```.


- Then, need to update this:
https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/components/ValuePicker/ValueSelectorModal.tsx#L29
to:


```
            <ScreenWrapper
                shouldEnableKeyboardAvoidingView={shouldEnableKeyboardAvoidingView}
```

- Similar with [ValuePicker](https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/components/ValuePicker/index.tsx), we also need to create a param ```shouldEnableKeyboardAvoidingView``` and default value is ```true```.

- Then update:

https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/components/ValuePicker/index.tsx#L48

```
            <ValueSelectorModal
                shouldEnableKeyboardAvoidingView={shouldEnableKeyboardAvoidingView}
```

- Finally, when use the ValuePicker component in [here](https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/pages/workspace/WorkspaceNewRoomPage.tsx#L296), [here](https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/pages/workspace/WorkspaceNewRoomPage.tsx#L307) and [here](https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/pages/workspace/WorkspaceNewRoomPage.tsx#L318), pass ```shouldEnableKeyboardAvoidingView={false}```.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by truph01** [comment]
### [Proposal updated](https://github.com/Expensify/App/issues/54157#issuecomment-2543143571)

**Comment by truph01** [comment]
Yeah, I think it should be fixed in https://github.com/Expensify/App/pull/53893#issuecomment-2543173308 since it has just been merged for 13 hours. Asked the PR author to check.

**Comment by greg-schroeder** [comment]
Awaiting results of the above, removing `Help Wanted`. Can you confirm https://github.com/Expensify/App/issues/54157#issuecomment-2543173745 @DylanDylann?

**Comment by DylanDylann** [comment]
This bug is fixed by https://github.com/Expensify/App/issues/53327. Let's close this one

**Comment by DylanDylann** [comment]
Calm down Melb, I need to check the above comment and evaluate this issue again

**Comment by DylanDylann** [comment]
@truph01 How about this KeyboardAvoidingView? Do we need to remove it?

https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/pages/workspace/WorkspaceNewRoomPage.tsx#L254-L260

**Comment by DylanDylann** [comment]
Yeah, but I worry that this KeyboardAvoidingView will affect the [ValuePicker](https://github.com/Expensify/App/blob/e5d7da88dde7e725ba9f2a919ad1df868040d719/src/components/ValuePicker/index.tsx) component

**Comment by DylanDylann** [comment]
Thanks @truph01 for pointing out that. 

First of all, I verify that this bug can still be reproduced although the blank area at the bottom of the screen seems smaller than on OP. 

@abzokhattab and @truph01 have the same solution for this problem. But I prefer to go with @truph01's [proposal](https://github.com/Expensify/App/issues/54157#issuecomment-2543143571) because they deep dive into the RCA and make the bug become more visible 

🎀 👀 🎀 C+ Reviewed

**Comment by DylanDylann** [comment]
@abzokhattab Thanks for your concern. I agree that the two solutions are the same. But I prefer @truph01's proposal because they have a detailed RCA that I don't see in your proposal. The decision to approve a proposal relies not only on the code change but also on how contributors explain and have a solid understanding of that bug. In this issue, I think @truph01 gives more value to their proposal.

Anyway, I appreciate your contribution and let's wait the internal engineer give the final decision.

**Comment by greg-schroeder** [comment]
Back from the holidays, apologies for the delay.

 It sounds like we need you to weigh in here @youssef-lr. What do you think of this [comment](https://github.com/Expensify/App/issues/54157#issuecomment-2560956928)?

**Comment by greg-schroeder** [comment]
Bump @youssef-lr on the above

**Comment by youssef-lr** [comment]
Not overdue.

**Comment by greg-schroeder** [comment]
Payment date is not 'til tomorrow

**Comment by DylanDylann** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by greg-schroeder** [comment]
Apologies, I was out sick. Processing

**Comment by greg-schroeder** [comment]
Payment summary:

Contributor: @truph01 - $250 - Upwork
C+: @DylanDylann - $250 - Upwork

**Comment by greg-schroeder** [comment]
Filed regression test
